File: Udemy/Algoritmos e Logica de Programacao/Java/ExerciciosJavaVetores/src/application/ProblemaDadosPessoas.java
package application;

import java.util.Locale;
import java.util.Scanner;

public class ProblemaDadosPessoas {

	public static void main(String[] args) {
		/*
		 * Problema "dados_pessoas"
Tem-se um conjunto de dados contendo a altura e o gênero (M, F) de N pessoas. Fazer um programa
que calcule e escreva a maior e a menor altura do grupo, a média de altura das mulheres, e o número
de homens.
		 * */

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		System.out.print("Quantas pessoas serao digitadas? ");
		int n = sc.nextInt();
		double[] altura = new double[n];
		char[] genero = new char[n];
		double maior =0.0;
		double menor = 0.0;
		double mediaMulher = 0.0;
		double somaMulher = 0.0;
		int contMulher = 0;
		int contHomem = 0;
		for(int i=0;i<n;i++) {
			System.out.print("Altura da "+(i+1)+"a pessoa: ");
			altura[i] = sc.nextDouble();
			System.out.print("Genero da "+(i+1)+"a pessoa: ");
			genero[i] = sc.next().charAt(0);
			if((genero[i]=='F')||(genero[i]=='f')) {
				somaMulher = somaMulher + altura[i];
				contMulher++;
			}else if((genero[i]=='M')||(genero[i]=='m')) {
				contHomem++;
			}
		}
		
		maior = altura[0];
		menor = altura[0];
		
		for(int i=0;i<n;i++) {
			if(altura[i]>maior) {
				maior = altura[i];
			}
			
			if(altura[i]<menor) {
				menor = altura[i];
			}
		}
		
		mediaMulher = (double)somaMulher / contMulher;
		
		System.out.println("Menor altura = "+String.format("%.2f", menor));
		System.out.println("Maior altura = "+String.format("%.2f", maior));
		System.out.println("Media das alturas das mulheres = "+String.format("%.2f", mediaMulher));
		System.out.println("Numero de homens = "+contHomem);
		
		sc.close();
	}

}

File: Algoritmos e Logica de Programacao/C++/Vetores/ProblemaDadosPessoas.cpp
/*
Problema "dados_pessoas"
Tem-se um conjunto de dados contendo a altura e o gênero (M, F) de N pessoas. Fazer um programa
que calcule e escreva a maior e a menor altura do grupo, a média de altura das mulheres, e o número
de homens.
*/



#include <iostream>
#include <iomanip>

using namespace std;

int main(){
	
	cout << "Quantas pessoas serao digitadas? ";
	int n=0;
	int i=0;
	cin >> n;
	double altura[n];
	char genero[n];
	double menor = 0.0, maior = 0.0;
	double mediaMulher = 0.0;
	double somaMulher = 0.0;
	int contMulher = 0;
	int contHomem = 0;
	for(i=0;i<n;i++){
		cout << "Altura da " << i+1<< "a pessoa: ";
		cin >> altura[i];
		cout << "Genero da "<< i+1 << "a pessoa: ";
		cin >> genero[i];
		if((genero[i] == 'F')||(genero[i] =='f')){
			somaMulher = somaMulher + altura[i];
			contMulher = contMulher+1;
		}else if((genero[i] == 'M')||(genero[i] =='m')){
			contHomem = contHomem + 1;
		}
	}
	
	menor = altura[0];
	maior = altura[0];
	
	for(i=0;i<n;i++){
		if(altura[i]>maior){
			maior = altura[i];
		}
		
		if(altura[i]<menor){
			menor = altura[i];
		}
				
	}
	
	mediaMulher = (double)somaMulher / contMulher;
	
	cout << fixed << setprecision(2);
	
	cout << "Menor altura = " << menor << endl;
	cout << "Maior altura = " << maior <<endl;
	cout << "Media das alturas das mulheres = " << mediaMulher << endl;
	cout << "Numero de homens = " << contHomem;

    return 0;
}
